Source/Layout/ASLayout.mm

```diff
@@ -216,28 +227,40 @@ - (ASLayout *)filteredNodeLayoutTree
   
   // Queue used to keep track of sublayouts while traversing this layout in a DFS fashion.
   std::deque<Context> queue;
-  queue.push_front({self, CGPointZero});
+  for (ASLayout *sublayout in self.sublayouts) {
+    queue.push_back({sublayout, sublayout.position});
+  }
+  
+  NSMutableArray *flattenedSublayouts = [NSMutableArray array];
   
   while (!queue.empty()) {
-    Context context = queue.front();
+    const Context context = queue.front();
     queue.pop_front();
-
-    if (self != context.layout && context.layout.type == ASLayoutElementTypeDisplayNode) {
-      ASLayout *layout = [ASLayout layoutWithLayout:context.layout position:context.absolutePosition];
-      layout.flattened = YES;
-      [flattenedSublayouts addObject:layout];
-    }
     
-    std::vector<Context> sublayoutContexts;
-    for (ASLayout *sublayout in context.layout.sublayouts) {
-      if (sublayout.isFlattened == NO) {
+    ASLayout *layout = context.layout;
+    const NSArray<ASLayout *> *sublayouts = layout.sublayouts;
+    const NSUInteger sublayoutsCount = sublayouts.count;
+    const CGPoint absolutePosition = context.absolutePosition;
+    
+    if (ASLayoutIsDisplayNodeType(layout)) {
+      if (sublayoutsCount > 0 || CGPointEqualToPoint(absolutePosition, layout.position) == NO) {
+        // Only create a new layout if the existing one can't be reused, which means it has either some sublayouts or an invalid absolute position.
+        layout = [ASLayout layoutWithLayoutElement:layout.layoutElement
```

Comment:
Is this the "aggressive reuse"? Is this case skipped often enough in practice to make a difference? Not sure of the performance impact of removing the flag but do we have a good guess as to the consequences of landing this PR?